### PR TITLE
Update to latest sparse simulator

### PR DESCRIPTION
**Pull request by swernli**
Pulls in fixes from https://github.com/qir-alliance/qir-runner/pull/186

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -69,7 +69,7 @@ wasm-bindgen-futures = "0.4"
 rand = "0.8"
 serde_json = "1.0"
 pyo3 = "0.20"
-quantum-sparse-sim = { git = "https://github.com/qir-alliance/qir-runner", rev = "54f2ffc097861146a2cc6595edaaffa7e2fdbf15", default-features = false }
```
Comment (billti) [inline Cargo.toml:72]: I'm guessing the removal of `default-features = false` was deliberate (doesn't appear to be any optional features on it).

**Review by billti — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (swernli) [reply Cargo.toml:72]: Yup, intentional. As of https://github.com/qir-alliance/qir-runner/pull/164 there is no longer any default/optional dependencies in the simulator, so while it has no effect it's clearer to remove it.

**Review by ScottCarda-MS — APPROVED** [review]

**Review by cesarzc — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -69,7 +69,7 @@ wasm-bindgen-futures = "0.4"
...
+quantum-sparse-sim = { git = "https://github.com/qir-alliance/qir-runner", rev = "2fac7354f4e98c3efc05f8fd4180a02d0b712907" }
```
Comment (idavis) [inline Cargo.toml:72]: Should this be main branch and not rev?

Comment (swernli) [reply Cargo.toml:72]: it's not part of a published release yet, and that rev is from the main branch (see https://github.com/qir-alliance/qir-runner/commits/main/)

**Review by idavis — APPROVED** [review]

**Pull request merged**